snap/quota/resources.go

```diff
@@ -124,6 +143,30 @@ func (qr *Resources) validateThreadQuota() error {
 	return nil
 }
 
+func (qr *Resources) validateJournalQuota() error {
+	// Journal quota is a bit different than the rest, we allow nil values
+	// for the size and rate, because this means that the only 'quota' we want
+	// in this case is to scope the journal for snaps in the group to just a namespace
+	// without any extra limits.
+	if qr.Journal.Size != nil {
+		if qr.Journal.Size.Limit == 0 {
+			return fmt.Errorf("journal size quota must have a limit set")
+		}
+
+		// max supported in systemd is 4GB
+		if qr.Journal.Size.Limit > 4*quantity.SizeGiB {
```

Comment:
Can you expose this limit (as well as 64kB min) as consts at the top?

Reply:
Done, thanks for the suggestion!